examples/calculating_time_lags.py

```diff
@@ -0,0 +1,141 @@
+# narrowband channels and thus two temperatures. To find this,
+# we can compute the cross-correlation between the two signals
+# and find which "lag" yields the highest correlation.
+
+lags = get_lags(time)
+cc = cross_correlation(s_a, s_b, lags)
+plt.plot(lags, cc)
+plt.xlabel("Lag [s]")
+plt.ylabel("Cross-correlation, AB")
+plt.show()
+
+###################################################################
+# Additionally, we can also easily calculate the maximum value of the
+# cross-correlation and the associate lag, or the time lag.
+
+tl = time_lag(s_a, s_b, time)
+max_cc = max_cross_correlation(s_a, s_b, time)
+plt.plot(lags, cc)
+plt.plot(tl, max_cc, marker="o", ls="", markersize=4)
+plt.xlabel("Lag [s]")
+plt.ylabel("Cross-correlation, AB")
+plt.show()
+
+###################################################################
+# As expected from the first intensity plot, we find that the lag
+# which maximizes the cross-correlation is approximately the separation
+# between the mean values of the Gaussian pulses.
+
+print("Time lag, A -> B = ", tl)
+
+###################################################################
+# Note that a positive time lag indicates that signal A has to be
+# shifted forward in time to match signal B. By reversing the order
+# of the inputs, we also reverse the sign of the time lag.
+
+print("Time lag, B -> A =", time_lag(s_b, s_a, time))
+
+###################################################################
+# The real power in the time lag approach is it's ability to reveal
+# large scale patterns of cooling in images of the Sun, particularly
+# in active regions. All of these functions can also be applied to
+# intensity data cubes to create a "time lag map".
+#
+# As an example, we'll create a fake data cube by repeating Gaussian
+# pulses with varying means and then add some noise to them
+
+means_a = np.tile(np.random.rand(10, 10), time.shape + (1, 1)) * u.s
+means_b = np.tile(np.random.rand(10, 10), time.shape + (1, 1)) * u.s
+noise = 0.2 * (-0.5 + np.random.rand(*means_a.shape))
+s_a = gaussian_pulse(np.tile(time, means_a.shape[1:] + (1,)).T, means_a, 0.02 * u.s) + noise
+s_b = gaussian_pulse(np.tile(time, means_b.shape[1:] + (1,)).T, means_b, 0.02 * u.s) + noise
+
+###################################################################
+# We can now compute a map of the time lag and maximum cross correlation.
+
+max_cc_map = max_cross_correlation(s_a, s_b, time)
+tl_map = time_lag(s_a, s_b, time)
+fig = plt.figure(figsize=(10, 5))
+ax = fig.add_subplot(121)
+im = ax.imshow(tl_map, cmap="RdBu", vmin=-1, vmax=1)
+cax = make_axes_locatable(ax).append_axes("right", size="5%", pad="1%")
+cb = fig.colorbar(im, cax=cax)
+cb.set_label(r"$\tau_{AB}$ [s]")
+ax = fig.add_subplot(122)
+im = ax.imshow(max_cc_map, vmin=0, vmax=1, cmap="magma")
+cax = make_axes_locatable(ax).append_axes("right", size="5%", pad="1%")
+cb = fig.colorbar(im, cax=cax)
+cb.set_label(r"Max cross-correlation")
+plt.tight_layout()
+plt.show()
```

Comment:
@nabobalis will probably disagree with me here, but there are multiple logical sections to this block:
```suggestion
max_cc_map = max_cross_correlation(s_a, s_b, time)
tl_map = time_lag(s_a, s_b, time)

fig = plt.figure(figsize=(10, 5))

ax = fig.add_subplot(121)
im = ax.imshow(tl_map, cmap="RdBu", vmin=-1, vmax=1)
cax = make_axes_locatable(ax).append_axes("right", size="5%", pad="1%")
cb = fig.colorbar(im, cax=cax)
cb.set_label(r"$\tau_{AB}$ [s]")

ax = fig.add_subplot(122)
im = ax.imshow(max_cc_map, vmin=0, vmax=1, cmap="magma")
cax = make_axes_locatable(ax).append_axes("right", size="5%", pad="1%")
cb = fig.colorbar(im, cax=cax)
cb.set_label(r"Max cross-correlation")

plt.tight_layout()
plt.show()
```